manifests/init.pp

```diff
@@ -146,49 +158,55 @@
-  $cmdport                                                         = undef,
+  Optional[Stdlib::Port] $cmdport                                  = undef,
   $commandkey                                                      = 0,
-  $config                                                          = $chrony::params::config,
-  $config_template                                                 = $chrony::params::config_template,
-  $config_keys                                                     = $chrony::params::config_keys,
-  $config_keys_template                                            = 'chrony/chrony.keys.erb',
-  $chrony_password                                                 = 'xyzzy',
-  $config_keys_owner                                               = $chrony::params::config_keys_owner,
-  $config_keys_group                                               = $chrony::params::config_keys_group,
-  $config_keys_mode                                                = $chrony::params::config_keys_mode,
-  $config_keys_manage                                              = true,
-  $keys                                                            = [],
-  $local_stratum                                                   = 10,
-  $log_options                                                     = undef,
-  $package_ensure                                                  = 'present',
-  $package_name                                                    = 'chrony',
+  Stdlib::Unixpath $config                                         = $chrony::params::config,
+  String[1] $config_template                                       = $chrony::params::config_template,
+  Stdlib::Unixpath $config_keys                                    = $chrony::params::config_keys,
+  String[1] $config_keys_template                                  = 'chrony/chrony.keys.erb',
+  String[1] $chrony_password                                       = 'xyzzy',
+  Variant[Integer[0],String[1]] $config_keys_owner                 = $chrony::params::config_keys_owner,
+  Variant[Integer[0],String[1]] $config_keys_group                 = $chrony::params::config_keys_group,
+  Stdlib::Filemode $config_keys_mode                               = $chrony::params::config_keys_mode,
+  Boolean $config_keys_manage                                      = true,
+  Array[String[1]] $keys                                           = [],
+  Integer[1,15] $local_stratum                                     = 10,
+  Optional[String[1]] $log_options                                 = undef,
+  String[1] $package_ensure                                        = 'present',
+  String[1] $package_name                                          = 'chrony',
   Optional[String] $package_source                                 = undef,
   Optional[String] $package_provider                               = undef,
   $refclocks                                                       = [],
```

Comment:
I wasn't sure what I should do with some of the Array parameters.  In most cases, it looks as if the templates currently support both arrays and plain strings.  I think it'd be neater to just support Arrays, but that would be a breaking change, so I thought I'd ask first.

Reply:
A breaking change should be documented, otherwise people get suprised...
So please leave it as it is, we can get back to this on a later date, when everything is settled again.